internal/service/server/handlers.go

```diff
@@ -100,17 +122,34 @@ func (s *Server) GetMetricHandlerWithBody(res http.ResponseWriter, req *http.Req
 }
 
 func (s *Server) GetMetricsHandler(res http.ResponseWriter, req *http.Request) {
-	gauges, counters := s.MetricService.GetMetrics()
+	gauges, counters, err := s.MetricService.GetMetrics(req.Context())
+	if err != nil {
+		http.Error(res, err.Error(), http.StatusInternalServerError)
+	}
+
 	body := getMetricsString(gauges, counters)
 
 	res.Header().Set("Content-type", "text/html")
-	_, err := io.WriteString(res, body)
+	_, err = io.WriteString(res, body)
 	if err != nil {
 		http.Error(res, err.Error(), http.StatusInternalServerError)
 	}
 	res.WriteHeader(http.StatusOK)
 }
 
+func (s *Server) PingDatabaseHandler(res http.ResponseWriter, req *http.Request) {
+	ctx, cancel := context.WithTimeout(req.Context(), time.Second)
+	defer cancel()
+
+	err := s.MetricService.PingDatabase(ctx)
```

Comment:
SUGGESTION: this is a handler for observability. Usually, you can run it on another port.
So, I would use only connection here.